### ISSUE TITLE
[UI] [Visualization] Problems with the map visualizations 

### ISSUE THREAD

**Issue by Desvelao** [wazuh/wazuh-kibana-app#3919]
| Wazuh | Elastic | Rev  | Security |
| ----- | ------- | ---- | -------- |
| 4.x   | 7.x     | 4xxx | Basic, ODFE, Xpack |

| Browser |
| ------- |
| Chrome, Firefox, Safari, etc|

**Description**
The map visualization has some problems:
- it is not sometimes completely rendered
- the legend could appear duplicated

The problems were seen in Wazuh dashboard, we should review if they appear in Kibana.

**Tasks**
- [x] Research the causes of the problems
- [x] Review if we can apply some fix taking into account that the visualizations are rendered using a method exposed by the core of plugin platform. If not, we should consider removing the visualizations if the problems are persistent.

**Screenshots**
![image](https://user-images.githubusercontent.com/34042064/160113180-c314ff83-fb54-44b8-b86d-f794cabfc844.png)

**Comment by asteriscos** [comment]
Tried to reproduce the issue many many times and finally got an exception:

![Screenshot from 2022-03-31 20-23-48](https://user-images.githubusercontent.com/9343732/161124440-c555f6e8-2eaf-44c2-9434-a6aa62d7b5f7.png)

![Screenshot from 2022-03-31 20-24-15](https://user-images.githubusercontent.com/9343732/161124464-5ccd03a7-3e3c-4918-946f-364466bd0945.png)

The file is:
`/plugin:tileMap/src/plugins/tile_map/public/tile_map_visualization.js`

Still no pattern or cause was found. This behavior seems to be very "random" and hard to reproduce.

**Comment by asteriscos** [comment]
Trying to find the source of this issue with @Desvelao,  we analyzed the maps visualizations JSON definitions to check if any of the properties was misconfigured or deprecated. We also compared the visualization JSON properties with a newly created map in the _Visualization_ plugin saved object. 
This tests were of little help to reproduce and find the cause of this issue as everything seemed to work as expected.

![image](https://user-images.githubusercontent.com/9343732/161290585-c4de479a-78b0-40ab-a8cb-5a889dac2e3a.png)


Doing some research about the way the visualizations rendered, I realized while the visualization is loading its wrapper has `display:none` style. This causes the map to miscalculate its coordinates and causes this issue. The fix proposal is to change the map visibility without collapsing its wrapper in the loading state.

![image](https://user-images.githubusercontent.com/9343732/161285529-be116a88-128d-4b53-9b3d-f39c78f64ed5.png)
